src/lcm/LifeCycleActions.cc

```diff
@@ -1916,3 +2023,174 @@ void LifeCycleManager::trigger_updatesg(int sgid)
     });
 }
 
+/*  -------------------------------------------------------------------------- */
+/*  -------------------------------------------------------------------------- */
+
+void LifeCycleManager::trigger_updatevnet(int vnid)
+{
+    trigger([this, vnid]
+    {
+        int  vmid, rc;
+
+        auto vnet = vnpool->get(vnid);
+
+        if ( vnet == nullptr )
+        {
+            return;
+        }
+
+        // -------------------------------------------------------------------------
+        // Iterate over VNET VMs, networks at hypervisor are updated one by one
+        // -------------------------------------------------------------------------
+        do
+        {
+            bool is_error  = false;
+            bool is_tmpl   = false;
+            bool is_update = false;
+
+            rc = vnet->get_outdated(vmid);
+
+            if ( rc != 0 )
+            {
+                vnet->remove_template_attribute("VNET_UPDATE");
+
+                vnpool->update(vnet.get());
+                return;
+            }
+
+            vnpool->update(vnet.get());
+
+            vnet.reset();
+
+            if ( auto vm = vmpool->get(vmid) )
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

We observed that you are using unintended operations or assignments that could lead to logical errors, vulnerabilities, or undefined behavior. To prevent these issues, ensure that you follow best practices, such as using explicit comparisons and clear operator usage, and using parentheses to clarify intent. Learn more - https://cvefeed.io/cwe/detail/cwe-480-use-of-incorrect-operator